### ISSUE TITLE
HW1

### ISSUE THREAD

**Issue by Zijianlalala** [maplezzz/ML2017S_Hung-yi-Lee_HW#2]
```
def GD(X, Y, w, eta, iteration, lambdaL2):
    list_cost = []
    for i in range(iteration):
        hypo = np.dot(X, w)
        loss = hypo - Y
        cost = np.sum(loss ** 2) / len(X)
        list_cost.append(cost)
       # 求grad的时候，不需要*2吗
        grad = np.dot(X.T, loss) / len(X) + lambdaL2 * w
        w = w - eta * grad
    return w, list_cost
```

**Comment by szhyaogaoML** [comment]
我也不是很懂，这是直接从李宏毅老师那里转过来的，没来得及看，但我直觉应该是没错，也许常数不重要也许是和哪个地方约掉了，请再看看


------------------ 原始邮件 ------------------
发件人:&nbsp;"Zijian Wu"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月12日(星期四) 晚上10:04
收件人:&nbsp;"maplezzz/NTU_ML2017_Hung-yi-Lee_HW"<NTU_ML2017_Hung-yi-Lee_HW@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;[maplezzz/NTU_ML2017_Hung-yi-Lee_HW] HW1 (#2)



def GD(X, Y, w, eta, iteration, lambdaL2):     list_cost = []     for i in range(iteration):         hypo = np.dot(X, w)         loss = hypo - Y         cost = np.sum(loss ** 2) / len(X)         list_cost.append(cost)        # 求grad的时候，不需要*2吗         grad = np.dot(X.T, loss) / len(X) + lambdaL2 * w         w = w - eta * grad     return w, list_cost
